protostar/commands/test/test_runner.py

```diff
@@ -179,7 +180,20 @@ async def _invoke_test_cases(
         execution_state: TestExecutionState,
     ) -> None:
         for test_case in test_suite.test_cases:
-            test_case_runner_factory = TestCaseRunnerFactory(execution_state)
-            test_case_runner = test_case_runner_factory.make(test_case)
-            test_result = await test_case_runner.run()
+            test_result = await self._invoke_test_case(test_case, execution_state)
             self.shared_tests_state.put_result(test_result)
+
+    @staticmethod
+    async def _invoke_test_case(
+        test_case: TestCase, initial_state: TestExecutionState
+    ) -> TestResult:
+        state: TestExecutionState = initial_state.fork()
+
+        # TODO(mkaput): Remove this in favor of setting mode explicitly by cheatcodes in setup hooks.
+        state.config.mode = TestMode.infer_from_contract_function(
```

Comment:
```suggestion
        state.test_case_config.mode = TestMode.infer_from_contract_function(
```
`state.config` could be interpreted as some kind of `StateConfiguration`

Reply:
I will do this in separate PR